### PR TITLE
Add ingress class metrics

### LINKED ISSUE #1895: Adding metrics for IngressClass objects
**What would you like to be added**:
Metrics for [IngressClass](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-class) Kubernetes objects.

**Why is this needed**:
The `kubernetes.io/ingress.class` annotation is deprecated from kubernetes v1.22+. `IngressClass` object was introduced to replace the deprecated annotation. `IngressClass` ties Ingress Classes and Ingress Controllers together so having KSM metrics for it would allow users to track relations between Ingresses and Controllers in metrics.

**Describe the solution you'd like**
A set of metrics similar to `kube_storageclass_.*` would suffice:

```
kube_ingressclass_annotations
kube_ingressclass_created
kube_ingressclass_info
kube_ingressclass_labels
```

The `kube_ingressclass_info` metric needs to have a label reflecting the `spec.controller`  field of `IngressClass` object.

**Additional context**

### PR DESCRIPTION
**Pull request by kaitoii11**
**What this PR does / why we need it**:
Add metrics for Ingress Class object as optional resource

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1895

### CODE REVIEW

File: docs/ingressclass-metrics.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# IngressClass Metrics
+
+| Metric name| Metric type | Labels/tags | Status |
+| ---------- | ----------- | ----------- | ----------- |
+| kube_ingressclass_annotations | Gauge | `ingressclass`=&lt;ingressclass-name&gt; <br> `annotation_INGRESSCLASS_ANNOTATION`=&lt;INGRESSCLASS_ANNOTATION&gt; | EXPERIMENTAL |
+| kube_ingressclass_info | Gauge | `ingressclass`=&lt;ingressclass-name&gt; <br> `controller`=&lt;ingress-controller-name&gt; <br> | STABLE |
```
Comment (CatherineF-dev) [inline docs/ingressclass-metrics.md:6]: Maybe EXPERIMENTAL first.

File: internal/store/ingressclass.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	descIngressClassAnnotationsName     = "kube_ingressclass_annotations"
+	descIngressClassAnnotationsHelp     = "Kubernetes annotations converted to Prometheus labels."
+	descIngressClassLabelsName          = "kube_ingressclass_labels"
+	descIngressClassLabelsHelp          = "Kubernetes labels converted to Prometheus labels."
+	descIngressClassLabelsDefaultLabels = []string{"ingressclass"}
+)
+
+func ingressClassMetricFamilies(allowAnnotationsList, allowLabelsList []string) []generator.FamilyGenerator {
+	return []generator.FamilyGenerator{
+		*generator.NewFamilyGenerator(
```
Comment (CatherineF-dev) [inline internal/store/ingressclass.go:40]: Use NewFamilyGeneratorWithStability with metric level = basemetrics.ALPHA

File: internal/store/builder.go [hunk 3]
```diff
@@ -470,6 +471,10 @@ func (b *Builder) buildRoleBindingStores() []cache.Store {
 	return b.buildStoresFunc(roleBindingMetricFamilies(b.allowAnnotationsList["rolebindings"], b.allowLabelsList["rolebindings"]), &rbacv1.RoleBinding{}, createRoleBindingListWatch, b.useAPIServerCache)
 }
 
+func (b *Builder) buildIngressClassStores() []cache.Store {
+	return b.buildStoresFunc(ingressClassMetricFamilies(b.allowAnnotationsList["ingressclass"], b.allowLabelsList["ingressclass"]), &networkingv1.IngressClass{}, createIngressClassListWatch, b.useAPIServerCache)
```
Comment (dgrisonnet) [inline internal/store/builder.go:475]:
```suggestion
	return b.buildStoresFunc(ingressClassMetricFamilies(b.allowAnnotationsList["ingressclasses"], b.allowLabelsList["ingressclasses"]), &networkingv1.IngressClass{}, createIngressClassListWatch, b.useAPIServerCache)
```

File: internal/store/ingressclass_test.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+						"foo": "bar",
+					},
+				},
+				Spec: networkingv1.IngressClassSpec{
+					Controller: "controller",
+				},
+			},
+			Want: `
+					# HELP kube_ingressclass_annotations Kubernetes annotations converted to Prometheus labels.
+					# HELP kube_ingressclass_labels Kubernetes labels converted to Prometheus labels.
```
Comment (CatherineF-dev) [inline internal/store/ingressclass_test.go:87]: Update metrics description.

Comment (kaitoii11) [reply internal/store/ingressclass_test.go:87]: By update metrics description, do you mean to add a stability version?
So in this case, fix it like below?
```
# HELP kube_ingressclass_annotations [EXPERIMENTAL] Kubernetes annotations converted to Prometheus labels.
```
If so, it seems to me that only STABLE is included in the header right now.  
https://github.com/kubernetes/kube-state-metrics/blob/master/pkg/metric_generator/generator.go#L83-L102

Comment (CatherineF-dev) [reply internal/store/ingressclass_test.go:87]: Oh, just found it's consistent with configmap. Will close this comment.
```
# HELP kube_configmap_annotations Kubernetes annotations converted to Prometheus labels.`
```

**Pull request merged**